pages/accountLists/[accountListId]/contacts/[[...contactId]].page.tsx

```diff
@@ -47,12 +47,31 @@ const ContactsPage: React.FC = () => {
           }
         : {
             pathname: '/accountLists/[accountListId]/contacts/',
-            query: queryWithoutContactId,
+            query: { ...queryWithoutContactId },
           },
     );
     setContactDetailsId(id);
   };
 
+  const setSearchTerm = (searchTerm?: string) => {
+    const { contactId: _, ...queryWithoutContactId } = query;
+    push(
+      contactDetailsId
+        ? {
+            pathname: '/accountLists/[accountListId]/contacts/[contactId]',
+            query: {
+              ...queryWithoutContactId,
+              contactId: contactDetailsId,
+              searchTerm,
+            },
+          }
+        : {
+            pathname: '/accountLists/[accountListId]/contacts/',
+            query: { ...queryWithoutContactId, searchTerm },
+          },
+    );
```

Comment:
Let's get the contact id logic out of here.

```suggestion
    push({
      pathname,
      query: {
        ...query,
        searchTerm,
      },
    });
```

Needs
```ts
const { query, push, isReady, pathname } = useRouter();
```
above.

Setting the url could happen in the child...